### PR TITLE
🌱 Remove nolint usages for gocyclo, paralleltest, wrapcheck and gocognit

### PR DESCRIPTION
**Pull request by Ankitasw**
**What this PR does / why we need it**:
Remove  usage of following:
`//nolint:gocyclo`
`//nolint:gocognit`
`//nolint:paralleltest`
`//nolint:wrapcheck`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Part of #2384

### CODE REVIEW

File: pkg/services/govmomi/vcenter/clone.go [hunk 1]
```diff
@@ -46,7 +46,7 @@ const (
 // the virtual machine to be created on the vCenter, which can be resolved by waiting on the task reference stored
 // in VMContext.VSphereVM.Status.TaskRef.
 //
-//nolint:gocognit,gocyclo
+//nolint:gocognit
```
Comment (killianmuldoon) [inline pkg/services/govmomi/vcenter/clone.go:49]: For me locally we can also get rid of this

Comment (killianmuldoon) [reply pkg/services/govmomi/vcenter/clone.go:49]: Neither of these is enabled in our config

Comment (Ankitasw) [reply pkg/services/govmomi/vcenter/clone.go]: Missed this.

**Comment by chrischdi** [comment]
/lgtm

/hold cancel

Thanks!

**Pull request merged**